distribution/sc-proxy-dex/src/wrapped_farm_token_merge.rs

```diff
@@ -0,0 +1,331 @@
+    + nft_deposit::NftDepositModule
+    + wrapped_lp_token_merge::WrappedLpTokenMerge
+{
+    #[proxy]
+    fn locked_asset_factory_proxy(
+        &self,
+        to: Address,
+    ) -> sc_locked_asset_factory::Proxy<Self::SendApi>;
+
+    #[proxy]
+    fn farm_contract_merge_proxy(&self, to: Address) -> elrond_dex_farm::Proxy<Self::SendApi>;
+
+    #[endpoint(mergeWrappedFarmTokens)]
+    fn merge_wrapped_farm_tokens(
+        &self,
+        farm_contract: Address,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        let caller = self.blockchain().get_caller();
+        require!(
+            self.intermediated_farms().contains(&farm_contract),
+            "Invalid farm contract address"
+        );
+
+        self.merge_wrapped_farm_tokens_and_send(
+            &caller,
+            &farm_contract,
+            Option::None,
+            opt_accept_funds_func,
+        )
+    }
+
+    fn merge_wrapped_farm_tokens_and_send(
+        &self,
+        caller: &Address,
+        farm_contract: &Address,
+        replic: Option<WrappedFarmToken<Self::BigUint>>,
+        opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        let deposit = self.nft_deposit(caller).get();
+        require!(!deposit.is_empty() || replic.is_some(), "Empty deposit");
+
+        let wrapped_farm_token_id = self.wrapped_farm_token_id().get();
+        self.require_all_tokens_are_wrapped_farm_tokens(&deposit, &wrapped_farm_token_id)?;
+
+        let mut tokens = self.get_wrapped_farm_tokens_from_deposit(&deposit)?;
+
+        if replic.is_some() {
+            tokens.push(replic.unwrap());
+        }
+        self.require_wrapped_farm_tokens_from_same_farm(&tokens)?;
+
+        let merged_farm_token_amount = self.merge_farm_tokens(farm_contract, &tokens);
+        let farming_token_amount = self.merge_farming_tokens(&tokens)?;
+        self.burn_deposit_tokens(caller, &deposit);
+
+        let new_attrs = WrappedFarmTokenAttributes {
+            farm_token_id: merged_farm_token_amount.token_id,
+            farm_token_nonce: merged_farm_token_amount.token_nonce,
+            farm_token_amount: merged_farm_token_amount.amount.clone(),
+            farming_token_id: farming_token_amount.token_id,
+            farming_token_nonce: farming_token_amount.token_nonce,
+            farming_token_amount: farming_token_amount.amount,
+        };
+
+        self.nft_create_tokens(
+            &wrapped_farm_token_id,
+            &merged_farm_token_amount.amount,
+            &new_attrs,
+        );
+        let new_nonce = self.increase_wrapped_farm_token_nonce();
+
+        self.send_nft_tokens(
+            &wrapped_farm_token_id,
+            new_nonce,
+            &merged_farm_token_amount.amount,
+            caller,
+            &opt_accept_funds_func,
+        )?;
+
+        Ok(())
+    }
+
+    fn require_deposit_empty_or_tokens_are_wrapped_farm_tokens(&self) -> SCResult<()> {
```

Comment:
This function is not used anymore?

Reply:
It's used in 3 other places in proxy_farm.rs